chipsec/hal/mmio.py

```diff
@@ -216,21 +209,22 @@ def get_MMIO_BAR_base_address(self, bar_name, bus=None):
                 try:
                     base = self.cs.read_register_field(bar_reg, base_field, preserve, bus=_bus)
                 except CSReadError:
-                    if self.logger.HAL:
-                        self.logger.log('[mmio] Unable to determine MMIO Base.  Using Base = 0x0')
                     base = 0
+                    self.logger.log_hal(f'[mmio] Unable to determine MMIO Base.  Using Base = 0x{base:X}')
                 try:
                     reg_mask = self.cs.get_register_field_mask(bar_reg, base_field, preserve)
                 except CSReadError:
-                    if self.logger.HAL:
-                        self.logger.log('[mmio] Unable to determine MMIO Mask.  Using Mask = 0xFFFF')
                     reg_mask = 0xFFFF
+                    self.logger.log_hal(f'[mmio] Unable to determine MMIO Mask.  Using Mask = 0x{reg_mask:X}')
             else:
                 base = self.cs.read_register(bar_reg, bus=_bus)
                 reg_mask = self.cs.get_register_field_mask(bar_reg, preserve_field_position=preserve)
             if 'limit_field' in bar:
                 limit_field = bar['limit_field']
                 limit = self.cs.read_register_field(bar_reg, limit_field, bus=_bus)
+            else:
+                limit = 0
```

Comment:
You already set `limit = 0` on line 197. You don't really have to set it again in this case. 
Also, do we want to allow the use of a limit if `limit_field` isn't defined? e.g. if `align_bits` is set in `bar`.